packages/hash/api/src/model/entityType.model.ts

```diff
@@ -234,13 +234,25 @@ class __EntityType {
     return dbEntityTypes.map((entityType) => new EntityType(entityType));
   }
 
-  static async getEntityTypeParents(
-    client: DBClient,
-    params: { entityTypeId: string },
-  ) {
-    const dbEntityTypes = await client.getEntityTypeParents(params);
+  async getParentEntityTypes(client: DBClient) {
+    const parentSchema$ids = (
+      Array.isArray(this.properties?.allOf) ? this.properties.allOf : []
+    )
+      .filter(
+        (allOfEntry: any): allOfEntry is { $ref: string } =>
+          "$ref" in allOfEntry,
+      )
+      .map((allOfEntry) => allOfEntry.$ref);
+
+    const parentEntityTypes = await Promise.all(
+      parentSchema$ids.map((schema$id) =>
+        client.getEntityTypeBySchema$id({ schema$id }),
+      ),
+    );
```

Comment:
**Requested change:** I think we should throw an error when `getEntityTypeBySchema$id` returns `null` instead of filtering these items out of the returned array quietly. If an existing schema references a non-existent entity type I think we should treat it as a critical error.

Doing this would also mean you don't have to use `.flatMap` at the end of this function.
```suggestion
    const parentEntityTypes = await Promise.all(
      parentSchema$ids.map(async (schema$id) => {
        const parentEntityType = await client.getEntityTypeBySchema$id({
          schema$id,
        });

        if (!parentEntityType) {
          throw new Error("Critical: ");
        }

        return parentEntityType;
      }),
    );
```

Reply:
It does make sense to treat this as an error. In turn it means that a schema that uses and external URL as a $ref will always be treated as an error. Is that okay?

Reply:
That's a good point, we don't always have to treat this as an error - but I'd argue when we support external URLs (currently I don't think we do?) this logic would help remind us where we need to account for that, instead of silently not including the parent entity type in an array.